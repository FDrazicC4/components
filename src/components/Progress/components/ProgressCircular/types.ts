import { StyledComponent } from '@components/types';
import { Colors, Sizes } from '@components/Progress/types';


export type Variants = (
    | 'indeterminate'
    | 'determinate'
    | 'static'
);


export interface Props extends StyledComponent {

    /**
     * The overall display of the progress indicator.
     * @default 'indeterminate'
     */
    variant?: Variants,

    /**
     * The color of the progress indicator.
     * @default 'primary'
     */
    color?: Colors,

    /**
     * The size of the progress indicator.
     * @default 'inherit'
     */
    size?: Sizes,

    /**
     * The value of the progress indicator for the determinate and static variants. Value between 0 and 100.
     */
    value?: number,

}
