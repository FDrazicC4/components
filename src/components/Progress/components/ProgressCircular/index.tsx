import React from 'react';
import clsx from 'clsx';
import { CircularProgress as MuiProgressCircular } from '@material-ui/core';
import { Props } from './types';


/**
 * The `<Progress.Circular />` component renders a spinning loading indicator.
 */
export const ProgressCircular = React.forwardRef(({
    variant = 'indeterminate',
    color = 'primary',
    size = 'inherit',
    className,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const classNames = clsx(
        `brew-ProgressCircular--variant-${variant}`,
        `brew-ProgressCircular--color-${color}`,
        `brew-ProgressCircular--size-${size}`,
        className,
    );


    return (
        <MuiProgressCircular
            className={classNames}
            ref={ref}
            thickness={4}
            variant={variant}
            {...otherProps}
        />
    );


});


ProgressCircular.displayName = 'ProgressCircular';


export default ProgressCircular;
