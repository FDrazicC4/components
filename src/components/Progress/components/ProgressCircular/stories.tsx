import React from 'react';
import { select, number } from '@storybook/addon-knobs';
import Flag from '@components/Flag';
import Progress from '@components/Progress';
import { Colors, Sizes } from '@components/Progress/types';
import { Variants } from './types';


export default {
    component: Progress.Circular,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Stable|Feedback/Progress/Circular',
};


export const General = (): React.ReactElement => {

    const color: Colors = select('color', [
        'primary',
        'secondary',
        'success',
        'danger',
        'warning',
        'info',
        'light',
        'dark',
    ], 'primary');
    const size: Sizes = select('size', [
        'small',
        'medium',
        'large',
        'inherit',
    ], 'inherit');
    const variant: Variants = select('variant', ['indeterminate', 'determinate', 'static'], 'indeterminate');
    const value: number = number('value', 40);

    return (
        <Progress.Circular color={color} size={size} value={value} variant={variant} />
    );
};


export const Default = (): React.ReactElement => <Progress.Circular />;
export const Indeterminate = (): React.ReactElement => <Progress.Circular variant="indeterminate" />;
export const Determinate = (): React.ReactElement => <Progress.Circular value={30} variant="determinate" />;
export const Static = (): React.ReactElement => <Progress.Circular value={40} variant="static" />;
