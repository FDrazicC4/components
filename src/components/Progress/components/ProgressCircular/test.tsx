import React from 'react';
import renderer from 'react-test-renderer';
import ProgressCircular from '.';


describe('ProgressCircular', () => {

    it('Renders correctly with defaults', () => {
        const tree = renderer
            .create(<ProgressCircular />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly with property variants', () => {
        const tree = renderer
            .create((
                <React.Fragment>
                    <ProgressCircular color="primary" size="small" variant="indeterminate" />
                    <ProgressCircular color="secondary" size="medium" value={5} variant="determinate" />
                    <ProgressCircular color="tertiary" size="large" variant="static" />
                    <ProgressCircular color="success" />
                    <ProgressCircular color="error" />
                    <ProgressCircular color="warning" />
                    <ProgressCircular color="info" />
                </React.Fragment>
            ))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
