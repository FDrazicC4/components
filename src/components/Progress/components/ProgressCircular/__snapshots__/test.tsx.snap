// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressCircular Renders correctly with defaults 1`] = `
<div
  className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20}
      strokeWidth={4}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`ProgressCircular Renders correctly with property variants 1`] = `
Array [
  <div
    className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-primary brew-ProgressCircular--size-small MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    aria-valuenow={5}
    className="MuiCircularProgress-root brew-ProgressCircular--variant-determinate brew-ProgressCircular--color-secondary brew-ProgressCircular--size-medium MuiCircularProgress-colorPrimary"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "transform": "rotate(53.823deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={
          Object {
            "strokeDasharray": "125.664",
            "strokeDashoffset": "113.411px",
          }
        }
      />
    </svg>
  </div>,
  <div
    aria-valuenow={0}
    className="MuiCircularProgress-root brew-ProgressCircular--variant-static brew-ProgressCircular--color-tertiary brew-ProgressCircular--size-large MuiCircularProgress-colorPrimary MuiCircularProgress-static"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "transform": "rotate(-90deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleStatic"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={
          Object {
            "strokeDasharray": "125.664",
            "strokeDashoffset": "125.664px",
          }
        }
      />
    </svg>
  </div>,
  <div
    className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-success brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-danger brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-warning brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="MuiCircularProgress-root brew-ProgressCircular--variant-indeterminate brew-ProgressCircular--color-info brew-ProgressCircular--size-inherit MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20}
        strokeWidth={4}
        style={Object {}}
      />
    </svg>
  </div>,
]
`;
